eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -1,29 +1,25 @@
-parameters:
-  # Use this parameter if you want to override the .NET SDK set by global.json
-  - name: DotNetCoreVersion
-    type: string
-    default: ''
-
 steps:
-  # We set DOTNET_ROLL_FORWARD so that .NET assemblies targeting older versions of .NET can run on newer ones.
-  # See also:
-  # https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet#options-for-running-an-application
-  # https://learn.microsoft.com/en-us/dotnet/core/runtime-discovery/troubleshoot-app-launch?pivots=os-windows#required-framework-not-found
-  # https://aka.ms/dotnet/app-launch-failed
-  # https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=powershell
-  # https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=powershell
-  - pwsh: |
-      echo "##vso[task.setvariable variable=DOTNET_ROLL_FORWARD;]Major"
-    displayName: "Set DOTNET_ROLL_FORWARD to Major"
-  # https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/use-dotnet-v2?view=azure-pipelines
+
+  # We aim to keep .NET SDK coming from global.json to bethe newest, to enable us to use the latest & greatest,
+  # and allow us to gradually migrate our .NET sources to such version.
+  # About global.json: https://learn.microsoft.com/en-us/dotnet/core/tools/global-json
+  - task: UseDotNet@2 # About UseDotNet@2 task: https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/use-dotnet-v2?view=azure-pipelines
+    displayName: "Use .NET SDK from global.json"
+    retryCountOnTaskFailure: 3
+    inputs:
+      useGlobalJson: true
+
+  # We install .NET 6.0.x runtime in addition to .NET coming from global.json because most of our tools target 6.0.x.
+  # Once we migrate all tools to a newer .NET version, we should update this to install that version instead.
   - task: UseDotNet@2
-    displayName: "Use .NET SDK ${{ coalesce( parameters.DotNetCoreVersion, 'from global.json') }}"
+    displayName: "Use .NET runtime 6.0.x"
     retryCountOnTaskFailure: 3
     inputs:
-      ${{ if eq( parameters.DotNetCoreVersion, '') }}:
-        useGlobalJson: true
-      ${{ else }}:
-        version: ${{ parameters.DotNetCoreVersion }}
+      packageType: runtime
+      version: 6.0.x
+      # performMultiLevelLookup comes into play when given .NET executable target runtime is different
+      # than the installed .NET SDK. Without this, such runtime would not be found.
+      performMultiLevelLookup: true
```

Comment:
I was expecting to see this on the SDK install above. Did you try to verify if we need to explicitly have this step for the 6 runtime?

Reply:
Per our chat, these builds shows the behavior on `performMultiLevelLookup` for SDK:

([source](https://github.com/Azure/azure-sdk-tools/commit/6b2a4efe490eee3027b5ef9ce5a1c179b047b66d))

For 6.0 target (fails):
- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2177201&view=logs&j=951e43bb-1e6d-59fb-d232-79b6dbbdcd74&t=3c49b6ca-e856-54cd-c01f-bfd23f6f6d30&l=22

For 7.0 target (works):
- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2177200&view=results